core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -173,25 +176,24 @@ case class PSBT(
     * @param index index of the InputPSBTMap to add tx to
     * @return PSBT with added tx
     */
-  def addTransactionToInput(tx: Transaction, index: Int): PSBT = {
+  def addUTXOToInput(tx: Transaction, index: Int): PSBT = {
     require(
       index < inputMaps.size,
       s"index must be less than the number of input maps present in the psbt, $index >= ${inputMaps.size}")
     require(!inputMaps(index).isFinalized,
             s"Cannot update an InputPSBTMap that is finalized, index: $index")
 
+    val previousElements = inputMaps(index).elements
+    val txIn = transaction.inputs(index)
     val elements = {
```

Comment:
nit: you can get rid of the braces here as it is hitting an `if` with its own braces